make_count/make_count.tex

```diff
@@ -0,0 +1,293 @@
+        
+        \item \textup{(Domain-metric compatibility).} The domain \texttt{input\_domain} matches one of the possible domains listed in the definition of \texttt{input\_metric}, and likewise \texttt{output\_domain} matches one of the possible domains listed in the definition of \texttt{output\_metric}.
+        
+        \item \textup{(Stability guarantee).} For every input \texttt{u}, \texttt{v} drawn from the \texttt{input\_domain} and for every pair $(\din, \dout)$, where $\din$ is of type \texttt{u32} and $\dout$ is of type \texttt{TO} (see line \ref{line:din-dout-type} of the pseudocode), if \texttt{u}, \texttt{v} are $\din$-close under the \texttt{input\_metric} and $\texttt{stability\_relation(din, dout) = True}$, then \texttt{function(u)}, \texttt{function(v)} are $\dout$-close under the \texttt{output\_metric}.
+    \end{enumerate}
+\end{theorem}
+
+\begin{proof} \textbf{(Part 1 -- appropriate output domain).}
+In section \ref{sec:python-pseudocode}, we see that any value of type \texttt{TO} is in the \texttt{output\_domain}, and in line \ref{line:TO-output} of the Python-style pseudocode, we see that the \texttt{function} is always guaranteed to return a value of type \texttt{TO}. Therefore, since our output domain is any value of type \texttt{TO}, we see that $\function$ has the appropriate output domain \texttt{output\_domain}.
+
+Moreover, for some input vector \texttt{v} drawn from \texttt{input\_domain}, $\function$ either returns $\texttt{exact\_int\_cast(len(data), TO)}$, which will be of type \texttt{TO} by the definition of \texttt{exact\_int\_cast}; or, if the casting fails, it returns $\texttt{get\_max\_consecutive\_int(TO)}$ which, from our definition of $\texttt{get\_max\_consecutive\_int}$, will be of type \texttt{TO}. Therefore, since our output domain is always some value of type \texttt{TO}, we see that $\function$ has the appropriate output domain \texttt{output\_domain}.
+\end{proof}
+
+\question{Is the second paragraph in the proof above of ``Appropriate output domain'' necessary?}
+
+\begin{proof} \textbf{(Part 2 -- domain-metric compatibility).}
+
+The \texttt{input\_domain} is \texttt{VectorDomain(AllDomain(TIA))}. Because our \texttt{input\_metric} of \texttt{SymmetricDistance} is compatible with any domain of the form \\\texttt{VectorDomain(inner\_domain)}, and because \texttt{VectorDomain(AllDomain(TIA))} is of this form, we see that it is compatible with our \texttt{input\_metric} of \texttt{SymmetricDistance}.
+
+The \texttt{output\_domain} is \texttt{AllDomain(TO)}. Because our \texttt{output\_metric} of \texttt{SymmetricDistance} is compatible with any domain of the form \texttt{AllDomain(T)} where \texttt{T} has the trait \texttt{Sub(Output=T)}, and because \texttt{AllDomain(TO)} is of this form and has the necessary trait, we see that it is compatible with our \texttt{output\_metric} of \texttt{AbsoluteDistance}.

```

Comment:
SymmetricDistance->AbsoluteDistance(TO)  (also add "(TO)" at end of sentence).

Reply:
Does T need more traits for AbsoluteDistance(TO) metric? (like DistanceConstant(TO))

Reply:
My feeling is that T does not need more traits. This is because the absolute distance is never actually calculated. However, it makes sense that subtraction should be defined on the type because otherwise how could we have a notion for what an absolute distance on that type would look like?

I am not fully confident in this response. To everyone, please let me know if you think of a reason for more traits.